app/templates/application.hbs

```diff
@@ -25,32 +25,21 @@
 <div id="main" class="grid-x">
   <div class="cell large-4" id="cd-chooser">
     {{#mapbox-gl id='map' initOptions=(hash style='mapbox://styles/mapbox/light-v9' zoom=zoom center=(array lng lat)) mapLoaded=(action 'handleMapLoad') as |map|}}
-      {{#map.source data=model sourceId='cds' as |source|}}
-        {{source.layer layerType='fill' source='cds' paintOptions=(hash fill-opacity=0)}}
-        {{source.layer layerType='line' source='cds' paintOptions=(hash line-width=2 line-color='#ae561f')}}
-        {{source.layer layerType='symbol' source='cds' layoutOptions=cdLabels.layout paintOptions=cdLabels.paint}}
+      {{#map.source sourceId='cds' source=cdSource as |source|}}
+        {{source.layer layer=cdFillLayer}}
+        {{source.layer layer=cdLineLayer}}
+        {{source.layer layer=cdLabelLayer}}
       {{/map.source}}
 
-      {{map.call 'fitBounds' bounds}}
+      {{map.call 'fitBounds' mapState.bounds}}
       {{map.on 'click' (action 'handleClick')}}
       {{map.on 'mousemove' (action 'handleMouseover')}}
       {{map.on 'mouseout' (action 'handleMouseleave')}}
-
       {{#if mapState.currentlySelected}}
-        {{#map.source data=mapState.geom sourceId='currentlySelected' as |source|}}
-          {{source.layer  layerType='fill'
-                          source='currentlySelected'
-                          paintOptions=(hash fill-color='#ae561f' fill-opacity=0.2 fill-outline-color='#ae561f')}}
+        {{#map.source sourceId='currentlySelected' source=cdSelectedSource as |source|}}
+          {{source.layer layer=cdSelectedLayer}}
         {{/map.source}}
       {{/if}}
-
-      {{#if mouseoverLocation}}
```

Comment:
@chriswhong want to make sure removing tooltip was intentional - I remember some discussion about how it was redundant with the labels.